joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -270,7 +276,70 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
       RubyOperators.hashInitializer,
       DispatchTypes.STATIC_DISPATCH
     )
-    callAst(call, argumentAsts)
+    scope.popScope()
+    blockAst(block, List(callAst(call, argumentAsts)))
+  }
+
+  protected def astForAssociationHash(node: Association, tmp: String): List[Ast] = {
+    node.key match {
+      case rangeExpr: RangeExpression =>
+        val expandedList = generateStaticLiteralsForRange(rangeExpr).map(x => {
+          astForSingleKeyValue(x, node.value, tmp)
+        })
+
+        if (expandedList.nonEmpty) {
+          expandedList
+        } else {
+          List(astForSingleKeyValue(node.key, node.value, tmp))
```

Comment:
Another thing you can do if you're feeling fancy, is create a list with consing, e.g.,
```scala
astForSingleKeyValue(node.key, node.value, tmp) :: Nil
```